File: Problems/FixTheCode/FTC_3_sorting-algorithms/java/source/BubbleSorter.java
public class BubbleSorter implements Sorter{

    public BubbleSorter(){}

    public int[] Sort(int[] array){
        System.out.println("Sorting array using BUBBLE SORT");
        int[] sortedArray = array.clone();
        int arrayLength = sortedArray.length;

        for (int i = 1; i < arrayLength; i++) {
            for (int j = 0; j < arrayLength - i; j++) {
                if (sortedArray[j] < sortedArray[j+1]){
                    Arrays.SwapTwoNumbers(sortedArray, j, j+1);
                }
            }
        }

       return sortedArray;
    }
}

File: Problems/FixTheCode/FTC_3_sorting-algorithms/python/source/bubbleSorter.py
from customSorter import CustomSorter
from arrays import swapTwoNumbers

class BubbleSorter(CustomSorter):

    def sort(self):
        print("Sorting array using Bubble Sort")
        arrayLen = len(self.array)

        for i in range(arrayLen):
            for j in range(arrayLen - i):
                if (self.array[j] > self.array[j+1]):
                    swapTwoNumbers(self.array, j, j+1)